src/p2p/net_peerlist.h

```diff
@@ -214,6 +215,16 @@ namespace nodetool
     }
   }
   //--------------------------------------------------------------------------------------------------
+  inline
+  peerlist_entry peerlist_manager::get_nth_latest_peer(peers_indexed& peerlist, const size_t n)
```

Comment:
I personally think we need more such refactorings. Also thanks for adding the comment below.